src/CoreWCF.NetTcp/src/CoreWCF/Channels/Framing/ExceptionConvertingDuplexPipe.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO.Pipelines;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CoreWCF.Channels.Framing
+{
+    internal class ExceptionConvertingDuplexPipe : IDuplexPipe
+    {
+        public ExceptionConvertingDuplexPipe(IDuplexPipe innerDuplexPipe)
+        {
+            Input = new ExceptionConvertingPipeReader(innerDuplexPipe.Input);
+            Output = new ExceptionConvertingPipeWriter(innerDuplexPipe.Output);
+        }
+
+        public PipeReader Input { get; }
+        public PipeWriter Output { get; }
+
+        private class ExceptionConvertingPipeReader : PipeReader
+        {
+            private PipeReader _input;
+
+            public ExceptionConvertingPipeReader(PipeReader input) => _input = input;
+
+            public override void AdvanceTo(SequencePosition consumed) => _input.AdvanceTo(consumed);
+            public override void AdvanceTo(SequencePosition consumed, SequencePosition examined) => _input.AdvanceTo(consumed, examined);
+            public override void CancelPendingRead() => _input.CancelPendingRead();
+            public override void Complete(Exception exception = null) => _input.Complete(exception);
+            public override async ValueTask<ReadResult> ReadAsync(CancellationToken cancellationToken = default)
```

Comment:
@mconnew wanted to understand why another wrapper around LoggingDuplexPipe? Can't we move the try/catch to LoggingDuplexPipe ?

Reply:
LoggingDuplexPipe is only used in Debug builds